syntax = "proto3";
package me.sylver.marver.solomon;

service SolomonService {
    rpc GetImageEmbedding (GetImageEmbeddingRequest) returns (GetImageEmbeddingResponse) {}
    rpc DetectFaces (DetectFacesRequest) returns (DetectFacesResponse) {}
    rpc GetOCR (GetOCRRequest) returns (GetOCRResponse) {}
}

message GetImageEmbeddingRequest {
    oneof input {
        string file_path = 4;
        string text_input = 5;
    }
}

message GetImageEmbeddingResponse {
    Embedding embedding = 1;
}

message DetectFacesRequest {
    string file_path = 1;
}

message DetectFacesResponse {
    repeated Face faces = 1;
}

message Face {
    BoundingBox bounding_box = 1;
    Embedding embedding = 2;
    float confidence = 3;
}

message BoundingBox {
    float x1 = 1;
    float y1 = 2;
    float x2 = 3;
    float y2 = 4;
}

message Embedding {
    repeated float value = 1;
}

message GetOCRRequest {
    string file_path = 1;
}

message GetOCRResponse {
    repeated OCR results = 1;
}

message OCR {
    string text = 1;
    BoundingBox bounding_box = 2;
    float confidence = 3;
}